pkg/proxy/iptables/proxier.go

```diff
@@ -446,6 +450,9 @@ func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
 					// Turn on healthcheck responder to listen on the health check nodePort
 					healthcheck.AddServiceListener(serviceName.NamespacedName, info.healthCheckNodePort)
 				}
+			} else {
+				// Delete healthcheck responders, if any, previously listening for this service
+				healthcheck.DeleteServiceListener(serviceName.NamespacedName, 0)
```

Comment:
are we unconditionally calling this for any/all services? are you sure it doesn't have side-effects for normal services? maybe we should do a periodic background gc instaed?